src/selectors/entities/schemes.js

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+// @flow
+
+import {getAllChannels} from 'selectors/entities/channels';
+import {getTeams} from 'selectors/entities/teams';
+import {ScopeTypes} from 'constants/schemes';
+import type {GlobalState} from 'types/store';
+import type {Scheme} from 'types/schemes';
+import type {Channel} from 'types/channels';
+import type {Team} from 'types/teams';
+
+export function getSchemes(state: GlobalState): { [string]: Scheme } {
+    return state.entities.schemes.schemes;
+}
+
+export function getScheme(state: GlobalState, id: string): Scheme {
+    const schemes = getSchemes(state);
+    return schemes[id];
+}
+
+export function getSchemeChannels(state: GlobalState, schemeId: string): Array<Channel> {
```

Comment:
Is this going to get called often or used in a `mapStatesToProps` function? If so, I'd suggest using `createSelector` [similar to this](https://github.com/mattermost/mattermost-redux/blob/master/src/selectors/entities/channels.js#L75) so that it can be memoized and not recalculated every time